xlators/mgmt/glusterd/src/glusterd-geo-rep.c

```diff
@@ -1111,10 +1086,7 @@ glusterd_gsync_volinfo_dict_set(glusterd_volinfo_t *volinfo, char *key,
 {
     int32_t ret = -1;
     char *gsync_status = NULL;
-    xlator_t *this = NULL;
-
-    this = THIS;
-    GF_ASSERT(this);
+    xlator_t *this = THIS;
```

Comment:
Same comment as above + it could have been passed from the single caller of this function - glusterd_set_gsync_knob() - which already does 'this = THIS'  (which by itself is called only from glusterd_set_gsync_confs() - which is called only from glusterd_op_gsync_create() which also does 'this = THIS;'

In short, lots of opportunities to clean this up. Not for this series of patches of course.

Reply:
Ack. Same as above..